docs/roadmap.md

```diff
@@ -0,0 +1,41 @@
+# 2019 Roadmap for druid / piet
+
+There's been a lot of activity in GUI and 2D graphics, but it's not easy to make sense of it all. This document tries to explain what's in place, what the plans are for the next few months, and also a grander dream of what it might become.
+
+## Smaller modules composed in layers
+
+Building a GUI is a large-scale task, and to make any headway it needs to be broken into smaller components. I've factored this work into a number of Rust crates, largely so that each can be worked on and improved separately.
+
+In addition, many of these crates are designed as common infrastructure, and could be used by other GUI efforts.
+
+### druid
+
+### druid-shell
+
+The current scope of druid-shell is similar to that of winit: it pops up a window, and takes inputs. There is one additional set of functionality: it gives access to platform menus.
+
+I didn't use winit because I wanted to experiment, and didn't know whether design decisions of winit would limit performance or functionality. Shortly I want to take a close look at those questions. I can imagine using winit for that common functionality and layering other functionality on top.
+
+### piet and back-ends
+
+I explained the motivation of piet in a [blog post](https://raphlinus.github.io/rust/graphics/2018/10/11/2d-graphics.html). Since then, it's coming along pretty nicely. There are three back-ends: Direct2D which is high performance and Windows-only, Cairo which is mature and portable, but not cleanly packaged as Rust crates and not very performant, and web.
+
+### As yet unnamed high-level text crate
+
+The biggest missing functionality in piet is text handling. What's there right now is very primitive. I think DirectWrite is a good model for what apps need, but of course it's Windows specific.
```

Comment:
I suppose you assume most people who care about this will know what directwrite does but writing them out might be nice...